### PR TITLE
feat: add S3 bucket for axe core reports and SNS queue for axe-core-urls

### PR DESCRIPTION
**Pull request by maxneuvians**
This PR adds an S3 bucket to the infrastructure to collect axe-core reports. It also adds a SNS topic that will send URLs to be scanned by axe-core.

### CODE REVIEW

File: terragrunt/aws/api/kms.tf [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+resource "aws_kms_key" "scan-websites" {
+  description         = "KMS Key"
+  enable_key_rotation = true
+
+  policy = <<EOF
+{
+  "Version" : "2012-10-17",
+  "Id" : "key-default-1",
+  "Statement" : [ {
+      "Sid" : "Enable IAM User Permissions",
```
Comment (CalvinRodo) [inline terragrunt/aws/api/kms.tf:13]: Instead of a HereDoc would be nice if this was a aws_policy_document

Comment (CalvinRodo) [reply terragrunt/aws/api/kms.tf:13]: Alternatively we could use `jsonencode()`

File: terragrunt/aws/api/s3.tf [hunk 2]
```diff
@@ -0,0 +1,14 @@
+resource "aws_s3_bucket" "axe-core-report-data" {
+  bucket = "${var.product_name}-${var.env}-axe-core-report-data"
```
Comment (CalvinRodo) [inline terragrunt/aws/api/s3.tf:2]: Should we append a random key to this?

**Review by CalvinRodo — CHANGES_REQUESTED** [review]
See my comments

File: terragrunt/aws/api/s3.tf (hunk 2 above)
Comment (maxneuvians) [reply terragrunt/aws/api/s3.tf:2]: Should be unique enough

File: terragrunt/aws/api/kms.tf (hunk 1 above)
Comment (maxneuvians) [reply terragrunt/aws/api/kms.tf]: using a `aws_iam_policy_document` now

**Review by CalvinRodo — APPROVED** [review]
:taco::cat:

**Pull request merged**